SubZero/src/model/filter/Filter.h

```diff
@@ -7,11 +7,42 @@
 
 #ifndef FILTER_H_
 #define FILTER_H_
+#include "DataDefinition.h"
+#include "RGBFilter.h"
+//#include "HSVFilter.h"
+
 
 class Filter {
+	/*
+	 * Filter is a virtual class that is further specified into specific
+	 * filters accessible by FilterFactory. In general filters will have
+	 * a filter() function that will filter input according to the
+	 * algorithm it contains and the parameters it holds.
+	 */
+
 public:
 	Filter();
 	virtual ~Filter();
+
+	// The filter function takes a pointer and applies the algorithm of the
+	// specific filter to the content of the pointer. The content of the
+	// pointer is altered and 0 is returned for successful filter, 1 for no
+	// action performed due to incorrect type, 2 for other errors.
+	virtual int filter(CamData * inputImg) {
+		return 1;
+	}
+	// Overload for FPGA filtering
+	virtual int filter(FPGAData * inputAttitude) {
+		return 1;
+	}
+
+	// Get the ID of the specific filter instance
+	std::string getID() {
+		return ID;
+	}
+private:
+	std::string ID;
+
```

Comment:
To my understanding (of what an ID's use is at least) the ID is used to try to uniquely identify different filters, so ideally when these are created, the creator assigns ID values to each and every instance of a filter (taking care to make each one unique).  To put it into your use case, they should have a different ID